openid-4-verifiable-presentations-1_0.md

```diff
@@ -432,54 +432,26 @@ Note: "https://self-issued.me/v2" is a symbolic string and can be used as an `au
 
 ## Verifier Metadata Management {#client_metadata_management}
 
-The `client_id_scheme` enables deployments of this specification to use different mechanisms to obtain and validate metadata of the Verifier beyond the scope of [@!RFC6749]. The term `client_id_scheme` is used since the Verifier is acting as an OAuth 2.0 Client.
+This specification defines additional mechanisms to enable Wallets to determine whether the provided `redirect_uri` and `response_uri` parameters and public verification keys belong to a particular Client, utilizing existing mechanisms such as Dynamic Client Registration [@!RFC7591], OpenID Federation [@!OpenID.Federation], and the `client_metadata` parameter defined by this specification. Additional Verifier Metadata is retrieved using existing mechanisms or the `client_metadata` parameter.
 
-This specification defines the following values for the `client_id_scheme` parameter, followed by the examples where applicable: 
+The Wallet MUST verify that the provided `redirect_uri` or `response_uri` belongs to the Client.
 
-* `pre-registered`: This value represents the [@!RFC6749] default behavior, i.e., the Client Identifier needs to be known to the Wallet in advance of the Authorization Request. The Verifier metadata is obtained using [@!RFC7591] or through out-of-band mechanisms.
+If the Authorization Request is signed, the Wallet MUST authenticate the Request Object against the `client_id` contained in the Request Object. In this case, any `jwks` parameter in the `client_metadata` parameter MUST NOT be used to verify the signature of the Request Object.
 
-* `redirect_uri`: This value indicates that the Verifier's Redirect URI (or Response URI when Response Mode `direct_post` is used) is also the value of the Client Identifier. The Authorization Request MUST NOT be signed. The Verifier MAY omit the `redirect_uri` Authorization Request parameter (or `response_uri` when Response Mode `direct_post` is used). All Verifier metadata parameters MUST be passed using the `client_metadata` parameter defined in (#vp_token_request).
+If the Verifier Metadata is not obtained from a trusted source and if the Authorization Request is unsigned, the Wallet MUST verify that the `client_id` matches the `redirect_uri` or `response_uri` parameter. Examples for trusted sources include prior registration (e.g., using [@!RFC7591]) or other out-of-band processes that allow the Wallet to retrieve trusted Verifier Metadata for a particular Client.
```

Comment:
It should be stated that this requirement only applies if the client_id is present.

Reply:
I'm not convinced this matters, isn't client_id mandatory in all cases where redirect_uri or response_uri can be used? (i.e. I think you can only omit client_id in unsigned browser API requests, where redirect_uri and response_uri aren't permitted to be used).